java/streaming/src/main/java/com/ray/streaming/api/context/RayContext.java

```diff
@@ -0,0 +1,52 @@
+package com.ray.streaming.api.context;
+
+import com.ray.streaming.api.stream.StreamSink;
+import com.ray.streaming.plan.Plan;
+import com.ray.streaming.plan.PlanBuilder;
+import com.ray.streaming.schedule.IJobSchedule;
+import com.ray.streaming.schedule.impl.JobScheduleImpl;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.ray.api.Ray;
+
+/**
+ * Ray Job Context.
+ */
+public class RayContext implements Serializable {
+
+  private transient AtomicInteger idGenerator;
+  private List<StreamSink> streamSinks;
+  private Plan plan;
+
+  private RayContext() {
+    this.idGenerator = new AtomicInteger(0);
+    this.streamSinks = new ArrayList<>();
+  }
+
+  public static RayContext buildContext() {
+    Ray.init();
+    return new RayContext();
+  }
+
+  /**
+   * construct streaming job dag && run streaming job
+   */
+  public void execute() {
+    PlanBuilder planBuilder = new PlanBuilder(this.streamSinks);
```

Comment:
yes, this is sink of the dataflow,  plan builder can traversal from sink to source to build logical plan

Reply:
yes, an executionNode will run on an actor, such as map/filter. and we use ray.call to implement send message to downstream

Reply:
Why traversing from sinks to sources and not from sources to sinks? Is it because you want to start the sources after starting the rest of the actors?

Reply:
I want to start all the actors at the same time. traversing from sinks to sources is only to build static DataFlow.

Reply:
we can also build DataFlow traversal from sources to sinks. But Maybe Some Source without sink